packages/server/api/controllers/tools.controller.js

```diff
@@ -48,6 +45,16 @@ const getToolById = async (id) => {
     if (tools.length === 0) {
       throw new Error(`incorrect entry with the id of ${id}`, 404);
     }
+    const categoriesList = await knex('categories')
+      .select('categories.name')
+      .join(
+        'tools_categories',
+        'tools_categories.category_id',
+        '=',
+        'categories.id',
+      )
+      .where('tools_categories.tool_id', '=', id);
+    tools = { ...tools, categories: categoriesList };
```

Comment:
```suggestion
    tools[0].categories = categoriesList;
```
You could do something like this here, so then you already have the categories in the tool object 👌🏽

Reply:
@shpomp Thanks. It helps a lot. Fixed that. ✅